src/Controls/src/Core/Application.cs

```diff
@@ -32,6 +33,7 @@ public Application()
 			_platformConfigurationRegistry = new Lazy<PlatformConfigurationRegistry<Application>>(() => new PlatformConfigurationRegistry<Application>(this));
 			// Initialize this value, when the app loads
 			_lastAppTheme = RequestedTheme;
+			MessagingCenter.Subscribe<IAppHostBuilder, IWindow>(this, nameof(IApplication.CreateWindow), (app, window) => _internalWindows.Add(window));
```

Comment:
See @mattleibow  this was the way i found to add windows to Maui.Control.Application, by using MessagingCenter, but i m not happy with this.. if you have a better idea ..